docs/design/events.rst

```diff
@@ -235,3 +235,30 @@ Parameters:
 -  **$value** The initial value of the plugin hook.
 
 .. warning:: The `$params` and `$value` arguments are reversed between the plugin hook handlers and trigger functions!
+
+
+Unregister Event/Hook Handlers
+------------------------------
+
+The functions ``elgg_unregister_event_handler`` and ``elgg_unregister_plugin_hook_handler`` can be used to remove
+handlers already registered by another plugin or Elgg core. The parameters are in the same order as the registration
+functions, except there's no priority parameter.
+
+.. code:: php
+
+    elgg_unregister_event_handler('login', 'user', 'myPlugin_handle_login');
+
+Anonymous functions or invokable objects cannot be unregistered, but dynamic method callbacks can be unregistered
```

Comment:
I still really don't like this, the idea that a plugin can register something that you can't possibly do anything to unregister...  It means any changes you wanted to make to the behavior would require forking the entire plugin.